packages/ui/AnnounceDecorator/useAnnounce.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+import Announce from './Announce';
+
+/**
+ * Object returned by `useAnnounce`
+ *
+ * @typedef {Object} useAnnounceInterface
+ * @property {Function} announce Called to alert the user of behavior for accessibility
+ * @property {Node}     children An additional element which must be added to the component tree to
+ *                               support notifying the user
+ * @private
+ */
+
+/**
+ * Provides a method to alert the user of behavior for accessibility.
+ *
+ * ```
+ * 	function Component () {
+ *		const {announce, children} = useAnnounce();
+ *
+ *		return (
+ *			<Base onClick={() => announce('Unusually important notification for accessibility')}>
+ *				{children}
+ *			</Base>
+ *		);
+ *	}
```

Comment:
I feel like this sample glosses over how to combine incoming children with `children` from `useAnnounce()`.  Perhaps the `children` prop could be renamed to not conflict with incoming children?  The whole DX around `Announce` isn't that great.  Could we improve it?  Does the child even need to be a child?  Could it be a sibling of the `Base` component in this case, since we can pass the announce separately?